Mathlib/Data/Matroid/Closure.lean

```diff
@@ -629,4 +631,115 @@ lemma ext_closure {M₁ M₂ : Matroid α} (h : ∀ X, M₁.closure X = M₂.clo
   eq_of_indep_iff_indep_forall (by simpa using h univ)
     (fun _ _ ↦ by simp_rw [indep_iff_forall_closure_diff_ne, h])
 
+
+section Spanning
+
+variable {S T I B : Set α}
+
+/-- A set is `spanning` in `M` if its closure is equal to `M.E`, or equivalently if it contains
+  a base of `M`. -/
+@[mk_iff]
+structure Spanning (M : Matroid α) (S : Set α) : Prop where
+  closure_eq : M.closure S = M.E
+  subset_ground : S ⊆ M.E
+
+attribute [aesop unsafe 10% (rule_sets := [Matroid])] Spanning.subset_ground
+
+lemma spanning_iff_closure (hS : S ⊆ M.E := by aesop_mat) : M.Spanning S ↔ M.closure S = M.E := by
+  rw [spanning_iff, and_iff_left hS]
+
+lemma closure_spanning_iff (hS : S ⊆ M.E := by aesop_mat) :
+    M.Spanning (M.closure S) ↔ M.Spanning S := by
+  rw [spanning_iff_closure, closure_closure, ← spanning_iff_closure]
+
+lemma spanning_iff_ground_subset_closure (hS : S ⊆ M.E := by aesop_mat) :
+    M.Spanning S ↔ M.E ⊆ M.closure S := by
+  rw [spanning_iff_closure, subset_antisymm_iff, and_iff_right (closure_subset_ground _ _)]
+
+lemma not_spanning_iff_closure (hS : S ⊆ M.E := by aesop_mat) :
+    ¬M.Spanning S ↔ M.closure S ⊂ M.E := by
+  rw [spanning_iff_closure, ssubset_iff_subset_ne, iff_and_self,
+    iff_true_intro (M.closure_subset_ground _)]
+  exact fun _ ↦ trivial
+
+lemma Spanning.superset (hS : M.Spanning S) (hST : S ⊆ T) (hT : T ⊆ M.E := by aesop_mat) :
+    M.Spanning T :=
+  ⟨(M.closure_subset_ground _).antisymm
+    (by rw [← hS.closure_eq]; exact M.closure_subset_closure hST), hT⟩
+
+lemma Spanning.closure_superset_eq (hS : M.Spanning S) (hST : S ⊆ T) : M.closure T = M.E := by
```

Comment:
I'm not sure what the best name is here, but I think this is what the conventions would suggest:
```suggestion
lemma Spanning.closure_eq_of_superset (hS : M.Spanning S) (hST : S ⊆ T) : M.closure T = M.E := by
```